src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToAsyncMethods.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(AnalyzeCompilationStart);
+        }
+
+        private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+        {
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingCancellationToken, out INamedTypeSymbol? cancellationTokenType))
+            {
+                return;
+            }
+
+            // Retrieve the ConfigureAwait methods that could also be detected, which can come from:
+            // - A Task
+            // - A generic Task
+            // - A ValueTask
+            // - A generic ValueTask
+            if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksTask, out INamedTypeSymbol? taskType) ||
+                !context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksValueTask, out INamedTypeSymbol? valueTaskType) ||
+                !context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksGenericTask, out INamedTypeSymbol? genericTaskType) ||
+                !context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksGenericValueTask, out INamedTypeSymbol? genericValueTaskType))
+            {
+                return;
+            }
+
+            const string configureAwaitName = "ConfigureAwait";
+            if (!(taskType.GetMembers(configureAwaitName).OfType<IMethodSymbol>().FirstOrDefault() is IMethodSymbol taskConfigureAwaitMethod) ||
+                !(genericTaskType.GetMembers(configureAwaitName).OfType<IMethodSymbol>().FirstOrDefault() is IMethodSymbol genericTaskConfigureAwaitMethod) ||
+                !(valueTaskType.GetMembers(configureAwaitName).OfType<IMethodSymbol>().FirstOrDefault() is IMethodSymbol valueTaskConfigureAwaitMethod) ||
+                !(genericValueTaskType.GetMembers(configureAwaitName).OfType<IMethodSymbol>().FirstOrDefault() is IMethodSymbol genericValueTaskConfigureAwaitMethod))
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(context =>
+            {
+                if (ShouldAnalyze(
+                    (IAwaitOperation)context.Operation,
+                    cancellationTokenType,
+                    taskConfigureAwaitMethod,
+                    genericTaskConfigureAwaitMethod,
+                    valueTaskConfigureAwaitMethod,
+                    genericValueTaskConfigureAwaitMethod,
+                    out IInvocationOperation? invocation))
+                {
+                    context.ReportDiagnostic(invocation!.CreateDiagnostic(ForwardCancellationTokenToAsyncMethodsRule));
+                }
+            },
+            OperationKind.Await);
```

Comment:
We should flag failure to propagate `CancellationToken` for more than just awaited invocations, as per our conversation in the original issue.

Reply:
Done. I also added unit tests that verify calls without an await. Please take a look whenever you get a chance.